include/sys/zfs_rlock.h

```diff
@@ -74,10 +77,15 @@ struct znode;
  */
 rl_t *zfs_range_lock_impl(zfs_rlock_t *zrl, uint64_t off, uint64_t len,
     rl_type_t type, struct znode *zp);
+#ifdef _KERNEL
 #define	zfs_range_lock(zp, off, len, type) \
 	zfs_range_lock_impl(&(zp)->z_range_lock, off, len, type, zp)
 #define	zvol_range_lock(zrl, off, len, type) \
 	zfs_range_lock_impl(zrl, off, len, type, NULL)
+#else
+#define	zfs_range_lock(zrl, off, len, type) \
+	zfs_range_lock_impl(zrl, off, len, type, NULL)
```

Comment:
You should just use zfs_range_lock_impl and don't bother with znode at all. Having different prototype of the "same" function for kernel and user is really ugly.